src/main/java/app/model/agents/TargetAgent.java

```diff
@@ -15,11 +15,10 @@ public class TargetAgent extends AgentImp
     private boolean guardDetected;
     private final boolean DEBUG = false;
 
-    public TargetAgent(Vector position, Vector direction, double radius, Type type, Ray targetRay, World world)
+    public TargetAgent(Vector position, Vector direction, double radius, Type type, Ray targetRay)
```

Comment:
This constructor is getting pretty ugly ...

Reply:
yeah, but TargetAgent needs the targetRay in order to function, so it needs to be a parameter.
I already deleted the World since it isn't used anywhere, but the targetRay has to stay.

Reply:
But the TargetAgent can see the target in the view?  And will need to update the target reference each time anyway, so why not just set this internally in the constructor instead?

Reply:
removed it